.travis.yml

```
@@ -62,7 +62,7 @@ matrix:
    env:
     - SKIP_MAKE=y
    before_script:
-    - $mingw64 cmake -DCMAKE_MAKE_PROGRAM=mingw32-make -G "MSYS Makefiles" -DCMAKE_TOOLCHAIN_FILE=./packaging/mingw-w64/mingw_toolchain.cmake \
+    - $mingw64 cmake -DCMAKE_MAKE_PROGRAM=mingw32-make -G "MSYS Makefiles" -DCMAKE_TOOLCHAIN_FILE=./packaging/mingw-w64/msys2_mingw_toolchain.cmake \
```

```suggestion
    - $mingw64 cmake -DCMAKE_MAKE_PROGRAM=mingw32-make -G "MinGW Makefiles" -DCMAKE_TOOLCHAIN_FILE=./packaging/mingw-w64/msys2_mingw_toolchain.cmake \
```
https://cmake.org/cmake/help/latest/generator/MinGW%20Makefiles.html
Pretty sure we should be using MinGW Makefiles, not MSYS Makefiles. MSYS is a POSIX emulation environment and creates a dependency on the MSYS DLL, where MinGW builds native Windows applications.

---

From memory I did initially try to generate MinGW Makefiles and could not get them to work under the MSYS2 shell. I now see they are intended to run under `cmd.exe`. There would probably be some work required to change approach and set up the path correctly for `cmd.exe`.

The cmake toolchain file I defined sets `MAKE`, `CC`, etc. so that it cross-compiles for MinGW. I will verify today that I haven't accidentally built against the MSYS runtime instead, but the fact that the unit tests run with only `/mingw64/bin/{libwinpthread-1.dll,libgcc_s_seh-1.dll,libstdc++-6.dll}` copied across suggests that it's working (unless MSYS has put its libraries in the system PATH in this build env).

So I guess in summary, I think this works already but if there's a much simpler way to do it then I'm open to it.

---

I forget exactly what happened when I switched it to "MSYS Makefiles" - for a while there I was changing all sorts of things and suspect a stale cache on Travis was giving me the same results every time.

"MinGW Makefiles" appears to work so long as the build runs in the MinGW64 shell rather than the MSYS2 shell, meaning the toolchain file is now only needed to cross-compile from linux.